jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/BuildDockerMojo.java

```diff
@@ -73,6 +75,9 @@ public void execute() throws MojoExecutionException, MojoFailureException {
     MavenProjectProperties projectProperties =
         MavenProjectProperties.getForProject(
             getProject(), getSession(), getLog(), tempDirectoryProvider);
+    UpdateChecker updateChecker =
```

Comment:
I still kinda think this would be cleaner if we skip wasn't part of the update checker mechanic and we only created an update checker as needed.

```
Optional<UpdateChecker> updaterChecker = MavenUpdateChecker.for(projectProperties, getLog())

...
...
updateChecker.ifPresent().finish....
```

Reply:
the  MojoCommon.VERSION_URL could be part of MavenUpdateChecker too

Reply:
or something like this (not necessarily this exactly)

Reply:
I'd rather not make a new class, but I agree the `Optional` approach could be better, so I'll try something like that.